Nodejs/Product/Nodejs/Project/NodeModulesNode.cs

```diff
@@ -27,9 +27,35 @@
 using Microsoft.VisualStudioTools.Project;
 using MessageBox = System.Windows.MessageBox;
 using Timer = System.Threading.Timer;
+using System.Threading.Tasks;
 
 namespace Microsoft.NodejsTools.Project {
     internal class NodeModulesNode : AbstractNpmNode {
+        /// <summary>
+        /// `INpmGlobalPackageProvider` that caches `IGlobalPackages` objects for reuse.
+        /// </summary>
+        private class CacheGlobalPackageProivder : INpmGlobalPackageProvider {
```

Comment:
nit: spelling, also move below usage. also keeping 'npm' in the name might make it easier to search for